scripts/aggregate-themes/aggregate-themes.js

```diff
@@ -0,0 +1,150 @@
+const OUTPUT = 'aggregated-themes.js';
+const OUTPUT_PATH = path.resolve(process.cwd());
+const DISCLAIMER = fs.readFileSync(path.resolve(__dirname, 'disclaimer.txt'), 'utf8');
+
+/**
+ * Aggregates theme assets into a single file.
+ *
+ * By default the theme will recursively search all dependencies.
+ * If a theme directory contains a root-theme.scss only that single file will be aggregated.
+ */
+class ThemeAggregator {
+  /**
+   * Aggregates theme assets.
+   * @returns {string|null} - The output path of the aggregated theme file. Null if not generated.
+   */
+  static aggregate() {
+    const defaultConfig = path.resolve(process.cwd(), CONFIG);
+
+    if (fs.existsSync(defaultConfig)) {
+      // eslint-disable-next-line global-require, import/no-dynamic-require
+      return ThemeAggregator.aggregateTheme(require(defaultConfig));
+    }
+
+    return null;
+  }
+
+  /**
+   * Aggregates theme assets.
+   * @param {Object} options - The aggregation options.
+   * @returns {string} - The output path of the aggregated theme file.
+   */
+  static aggregateTheme(options = {}) {
+    ThemeAggregator.validate(options);
+
+    const { theme } = options;
+
+    const dirs = ThemeAggregator.find(`**/themes/${theme}/`, options);
+    const assets = ThemeAggregator.filter(dirs, options);
+
+    if (fs.existsSync(`${NODE_MODULES}${theme}/`)) {
+      assets.unshift(theme);
+    }
+
+    return ThemeAggregator.writeFile(assets, options);
+  }
+
+  /**
+   * Filters theme assets.
+   * @param {string[]} patterns - An array of directory and file paths.
+   * @param {Object} options - The aggregation options.
+   * @returns {string[]} - An array of file names.
+   */
+  static filter(patterns, options) {
+    const assets = [];
+    patterns.forEach((asset) => {
+      if (fs.lstatSync(asset).isDirectory()) {
+        assets.push(...ThemeAggregator.filterDir(asset, options));
+      } else {
+        assets.push(asset);
+      }
+    });
+
+    return assets.map(asset => ThemeAggregator.resolve(asset, options));
+  }
+
+  /**
+   * Filters theme files within a directory.
+   * @param {string} dir - The directory path.
+   * @param {Object} options - The aggregation options.
+   * @returns {string[]} - An array of filtered file names.
+   */
+  static filterDir(dir, options) {
+    // Include only the root file if one exists and it is not excluded.
+    const rootFile = ThemeAggregator.find(`${dir}root-theme.scss`, options);
+    if (rootFile.length === 1) {
+      return rootFile;
+    }
+
+    return ThemeAggregator.find(`${dir}*.scss`, options);
+  }
+
+  /**
+   * Finds files and directories matching a pattern.
+   * @param {string} pattern - A regex pattern.
+   * @param {Object} options - The aggregation options.
+   * @returns {string[]} - An array of matching files and directories.
+   */
+  static find(pattern, options) {
+    const { exclude = [] } = options;
+
+    return glob.sync(pattern, { ignore: exclude });
```

Comment:
It'd be worth considering adding `follow` key to the glob options. This will allow the search patterns to follow symlinked folders